tests/cephfs/clients/multiple_clients_permission_mounted_directories.py

```diff
@@ -0,0 +1,108 @@
+        # create 3 directories to mount
+        client1.exec_command(sudo=True, cmd=f"mkdir -p {fuse_mounting_dir_1}")
+        client1.exec_command(sudo=True, cmd=f"mkdir -p {fuse_mounting_dir_2}")
+        client1.exec_command(sudo=True, cmd=f"mkdir -p {fuse_mounting_dir_3}")
+        # create 3 clients with different permissions
+        name1 = "name1"
+        name2 = "name2"
+        name3 = "name3"
+        # prepare 3 clients with different permissions
+        client1.exec_command(
+            sudo=True,
+            cmd=f"ceph auth get-or-create client.{name1} "
+            f"mds 'allow *,allow * path=/dir1' mon 'allow r' osd 'allow rw tag cephfs data=cephfs' "
+            f"-o /etc/ceph/ceph.client.{name1}.keyring",
+        )
+        client1.exec_command(
+            sudo=True,
+            cmd=f"ceph auth get-or-create client.{name2} "
+            f"mds 'allow *,allow r path=/dir1' mon 'allow r' osd 'allow rw tag cephfs data=cephfs' "
+            f"-o /etc/ceph/ceph.client.{name2}.keyring",
+        )
+        client1.exec_command(
+            sudo=True,
+            cmd=f"ceph auth get-or-create client.{name3} "
+            f"mds 'allow *,allow * path=/dir2' mon 'allow r' osd 'allow rw tag cephfs data=cephfs' "
+            f"-o /etc/ceph/ceph.client.{name3}.keyring",
+        )
+        # mount 3 clients
+        client1.exec_command(
+            sudo=True, cmd=f"ceph-fuse -n client.{name1} {fuse_mounting_dir_1}"
+        )
+        client1.exec_command(
+            sudo=True, cmd=f"ceph-fuse -n client.{name2} {fuse_mounting_dir_2}"
+        )
+        client1.exec_command(
+            sudo=True, cmd=f"ceph-fuse -n client.{name3} {fuse_mounting_dir_3}"
+        )
+        # create 2 dirs
+        client1.exec_command(sudo=True, cmd=f"mkdir -p {fuse_mounting_dir_1}dir1")
+        client1.exec_command(sudo=True, cmd=f"mkdir -p {fuse_mounting_dir_1}dir2")
+        # Test permissions with the clients
+        client1.exec_command(sudo=True, cmd=f"touch {fuse_mounting_dir_1}dir1/file1")
+        out_ls1 = client1.exec_command(sudo=True, cmd=f"ls {fuse_mounting_dir_1}dir1/")
+        log.info(f"ls output in {fuse_mounting_dir_1} is {out_ls1[0]}")
+        if "file1" not in out_ls1[0]:
+            raise CommandFailed(f"write failed in {fuse_mounting_dir_1} with client1")
+        client1.exec_command(sudo=True, cmd=f"touch {fuse_mounting_dir_2}dir1/file2")
+        out_ls2 = client1.exec_command(sudo=True, cmd=f"ls {fuse_mounting_dir_2}dir1/")
+        log.info(f"ls output in {fuse_mounting_dir_2} is {out_ls2[0]}")
+        if "file2" in out_ls2[0]:
+            raise CommandFailed(
+                f"file2 can not be written thru {fuse_mounting_dir_2} with client2"
+            )
+        client1.exec_command(sudo=True, cmd=f"touch {fuse_mounting_dir_1}dir2/file3")
+        out_ls3 = client1.exec_command(sudo=True, cmd=f"ls {fuse_mounting_dir_1}dir1/")
```

Comment:
Since the previous command tried writing to dir2, shouldn't the contents check via 'ls' be done on 'dir2' instead of 'dir1'?
out_ls3 = client1.exec_command(sudo=True, cmd=f"ls {fuse_mounting_dir_1}dir2/")